qemu/tests/migration_with_sgx.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+from virttest import error_context
+from provider.sgx import SGXTestCapability
+from virttest import env_process
+from virttest.utils_misc import verify_dmesg
+from provider.sgx import SGXChecker
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Do migration test with sgx enabled
+
+    1. Check sgx and numa capability
+    2. Boot src sgx guest with numa
+    3. Migration
+    4. check sgx epc size and policy
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    sgx_cap = SGXTestCapability(test, params)
+    host_epc_size = sgx_cap.valid_host_sgx_cap()
+    params["host_epc_size"] = host_epc_size
+    if "mem" in params["mem_devs"]:
+        sgx_cap.valid_host_numa_node()
+
+    params['start_vm'] = "yes"
+    vm_name = params["vms"]
+    error_context.context("Start to boot sgx source vm", logging.info)
+    env_process.preprocess_vm(test, params, env, vm_name)
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    verify_dmesg()
+    dmesg_output = session.cmd_output(params["guest_sgx_check"]).strip()
+    params['dmesg_output'] = dmesg_output
+    session.close()
+    test_check = SGXChecker(test, params, vm)
+    test_check.guest_dmsg_sgx_verify()
+
+    # do migration
+    error_context.context("Start to do sgx vm migration", logging.info)
+    mig_timeout = float(params.get("mig_timeout", "3600"))
```

Comment:
params.get_numeric("mig_timeout", 3600)

Reply:
Thanks, updated

Reply:
mig_timeout is not a float, we can use an int here